src/Workleap.DomainEventPropagation.Subscription.PullDelivery.Tests/EventPullerTests.cs

```diff
@@ -1,58 +1,258 @@
+            var options2 = GivenOption(optionMonitor, clientName2);
+
+            var eventBundle = AutoFaker.Generate<EventGridClientAdapter.EventGridClientAdapter.EventBundle>();
+            RegisterCloudEventsInClient(functionalClient, options2, eventBundle);
+            var eventHandler = A.Fake<ICloudEventHandler>();
+
+            // When
+            var puller = new EventPuller(eventGridClientDescriptors, clientFactory, eventHandler, optionMonitor, new NullLogger<EventPuller>());
+            await StartWaitAndStop(puller);
+
+            // Then
+            A.CallTo(() => failingClient
+                    .ReceiveCloudEventsAsync(
+                        options1.TopicName,
+                        options1.SubscriptionName,
+                        A<CancellationToken>._))
+                .MustHaveHappenedOnceOrMore();
+            A.CallTo(() => eventHandler.HandleCloudEventAsync(eventBundle.Event, A<CancellationToken>._)).MustHaveHappenedOnceOrMore();
+        }
+    }
+
+    public class OneSubscriber : EventPullerTests, IDisposable
+    {
+        private readonly EventPuller _puller;
+        private readonly IEventGridClientAdapter _client;
+        private readonly EventPropagationSubscriptionOptions _option;
+        private readonly ICloudEventHandler _eventHandler;
+        private readonly EventGridClientAdapter.EventGridClientAdapter.EventBundle _eventBundle1;
+
+        private readonly EventGridClientAdapter.EventGridClientAdapter.EventBundle _eventBundle2;
+
+        public OneSubscriber()
+        {
+            var clientName = AutoFaker.Generate<string>();
+            var eventGridClientDescriptors = new EventGridClientDescriptor[] { new(clientName) };
+
+            var optionMonitor = A.Fake<IOptionsMonitor<EventPropagationSubscriptionOptions>>();
+            this._option = GivenOption(optionMonitor, clientName);
+
+            var clientFactory = A.Fake<IEventGridClientWrapperFactory>();
+            this._client = GivenFakeClient(clientFactory, clientName);
+            this._eventBundle1 = AutoFaker.Generate<EventGridClientAdapter.EventGridClientAdapter.EventBundle>();
+            this._eventBundle2 = AutoFaker.Generate<EventGridClientAdapter.EventGridClientAdapter.EventBundle>();
+            RegisterCloudEventsInClient(this._client, this._option, this._eventBundle1, this._eventBundle2);
+
+            this._eventHandler = A.Fake<ICloudEventHandler>(opt => opt.Strict());
+
+            this._puller = new EventPuller(eventGridClientDescriptors, clientFactory, this._eventHandler, optionMonitor, new NullLogger<EventPuller>());
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                this._puller.Dispose();
+            }
+        }
```

Comment:
Is there any reason why those tests dispose of the puller whereas the other test classes don't?

Reply:
They do now